src/logic/context/ContextManager.ts

```diff
@@ -0,0 +1,65 @@
+import {Context} from "../../data/Context";
+import {HotKeyAction} from "../../data/HotKeyAction";
+import {store} from "../../index";
+import {updateActiveContext} from "../../store/general/actionCreators";
+import * as _ from "lodash";
+
+export class ContextManager {
+    private static activeCombo: string[] = [];
+    private static actions: HotKeyAction[] = [];
+
+    public static init(): void {
+        window.addEventListener("keydown", ContextManager.onDown);
+        window.addEventListener("keyup", ContextManager.onUp);
+    }
+
+    public static register(context: Context, actions: HotKeyAction[]): void {
+        store.dispatch(updateActiveContext(context));
+        ContextManager.actions = actions;
+    }
+
+    private static onDown(event: KeyboardEvent): void {
+        const keyCode: string = ContextManager.getKeyCodeFromEvent(event);
+        if (!ContextManager.isInCombo(keyCode)) {
+            ContextManager.addToCombo(keyCode);
+            ContextManager.execute(event);
+        }
+    }
+
+    private static onUp(event: KeyboardEvent): void {
+        const keyCode: string = ContextManager.getKeyCodeFromEvent(event);
+        ContextManager.removeFromCombo(keyCode);
+    }
+
+    private static execute(event: KeyboardEvent): void {
+        for (let i = 0; i < ContextManager.actions.length; i++) {
+            const hotKey: HotKeyAction = ContextManager.actions[i];
+            if (ContextManager.matchCombo(ContextManager.activeCombo, hotKey.keyCombo)) {
+                hotKey.action(event);
+            }
+        }
+    }
+
+    private static isInCombo(keyCode: string): boolean {
+        return ContextManager.activeCombo.indexOf(keyCode) >= 0;
+    }
+
+    private static addToCombo(keyCode: string): void {
```

Comment:
do zbierania takich rzeczy ja bym zrobił KeyboardAndMouseManager'a i KeyboardAndMouseModel (znajdź lepszą nazwę xD) 
KeyboardAndMouseManager.init() - robi addEventListenery na mouseDown, keyDown, itd...
a same handlery przypisane do tych eventów zbierają/usuwają info o tym co się dzieje na klawiaturze/myszce i zapisują do modelu.
jak to będzie reduxowy model, to będziesz mógł widoki zapinać, ale na razie zrób suchy model

to tak w skrócie

Reply:
dobra. dokończę tutaj :D 

jak już masz tego KeyboardAndMouseManagera i Model, i ContextManager'a to:
1. Robisz `_____Context.ts` czyli np: `EditorContext.ts`
2. W tych obiektach robisz `this.listen("ctrl + click", ()=>EditorAction.update(EditorModel.getEdiorData()))` itd dla innych (to jest pseudokod więc frywolnie podszedłem) Pamietasz jak mówiliśmy o rozdzieleniu tego Editor.tsx na Model i akcje? :D
ta metoda `listen` zapina się na wszystko na co została zapięta (jeśli akutalny context zapina się chociaż raz na click, to `listen` powinein się uruchamiać na każdy mouseDown, mouseUp, keyDown, itd...) i jeśli ZOSTANIE SPEŁNIONY WYMÓG (np: z powyższego przykładu) - czyli aktualnie jest wykonany click, oraz wciśnięty CTRL, to wykonaj zapiętą akcję
3. Gdzieś powinieneś rejestrować te konteksty 
4. No i najważniejsze to switchowanie tych kontekstów, no ale to już pozostawiam Tobie. po prostu trzeba znaleźć odpowiedni sposób na uruchamianie `ContextManager.switch(Contexts.EDITOR)`

ponownie w dużym uproszczeniu, bo nie opisałem wyrejestrowania `listen`ów itd. ale mam nadzieję że wiesz ocb